numpy/_core/src/multiarray/dtypemeta.cpp

```diff
@@ -1121,22 +1136,59 @@ dtypemeta_wrap_legacy_descriptor(
      * Any Type slots need to be fixed before PyType_Ready, although most
      * will be inherited automatically there.
      */
+    static const char *docstring =
+            "DType class corresponding to the scalar type and dtype of "
+            "the same name.\n\n"
+            "Please see `numpy.dtype` for the typical way to create\n"
+            "dtype instances and :ref:`arrays.dtypes` for additional\n"
+            "information.";
     static PyArray_DTypeMeta prototype = {
-        {{
-            PyVarObject_HEAD_INIT(&PyArrayDTypeMeta_Type, 0)
-            .tp_name = NULL,  /* set below */
-            .tp_basicsize = sizeof(_PyArray_LegacyDescr),
-            .tp_flags = Py_TPFLAGS_DEFAULT,
-            .tp_base = NULL,  /* set below */
-            .tp_new = (newfunc)legacy_dtype_default_new,
-            .tp_doc = (
-                "DType class corresponding to the scalar type and dtype of "
-                "the same name.\n\n"
-                "Please see `numpy.dtype` for the typical way to create\n"
-                "dtype instances and :ref:`arrays.dtypes` for additional\n"
-                "information."),
-        },},
-        .flags = NPY_DT_LEGACY,
+        {
+                {
+                        PyVarObject_HEAD_INIT(&PyArrayDTypeMeta_Type, 0)
+                                NULL, /* set below */
+                        sizeof(_PyArray_LegacyDescr),
+                        0,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        Py_TPFLAGS_DEFAULT,
+                        docstring,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL, /* tp_base set below */
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        NULL,
+                        (newfunc)legacy_dtype_default_new,
+                },
+        },
+        NULL,
+        0, /* type_num filled in below */
+        NULL,
+        NPY_DT_LEGACY,
```

Comment:
not sure if there's a less verbose way to deal with MSVC not liking designated initializer lists in C++ code.